### PR TITLE
4.06

### PR DESCRIPTION
**Pull request by moxhn**

### CODE REVIEW

File: .travis-ci.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@ case $XARCH in
 i386)
   uname -a
 
-  git clone git://github.com/ocaml/ocaml -b trunk --depth 1
+  git clone git://github.com/ocaml/ocaml -b 4.06 --depth 1
```
Comment (gasche) [inline .travis-ci.sh:5]: this change is probably unintentional, right? I think the intent was to track the compiler's `trunk`.

File: opam [hunk 2]
```diff
@@ -1,24 +1,31 @@
 opam-version: "1.2"
+version: "4.06+trunk"
```
Comment (gasche) [inline opam:2]: this part of the new `opam` file is probably to be replaced with `"trunk"` only.

**Comment by gasche** [comment]
It looks like we forgot to synchronize `trunk` with version-branches and it missed a lot of commits, and @moxhn is now proposing to merge the latest branch (4.06) back into trunk. This sounds reasonable to me. cc @diml, who may be able to comment on how the one-branch-per-version model is supposed to work, and @dra27 who can help us ensure that those things don't slip back in the future.

**Comment by ghost** [comment]
The one-branch-per-version model is a huge pain. It would be much better to have the same branch be compatible with all versions of the compiler, but this needs a bit of refactoring work.

**Comment by ghost** [comment]
BTW, one thing I wanted to do was use ocaml-migrate-parsetree in camlp4. This way we wouldn't have to update it  anymore

**Comment by XVilka** [comment]
@diml so this one can be closed then?

**Comment by ghost** [comment]
Yes, that seems right

**Pull request closed without merging**